src/DataWrangling/EN4/EN4_restoring.jl

```diff
@@ -0,0 +1,337 @@
+using Oceananigans: location
```

Comment:
I think also this file is basically general and can be reused.

Reply:
That's right, this is basically a complete copy/paste. The only thing I have done is removed the extra variables that don't exist in EN4, like velocities and sea ice thickness. 

I should note I haven't fully tested if the restoring is working so it may be that we need to come back and pull some functions out to work differently for EN4, but as it is this is basically identical.

Reply:
Should we try to get field working first in this PR, and then tackle restoring in a future PR?

Reply:
I agree with this suggestion.